### PR TITLE
SM can explode on space tiles again

### PR DESCRIPTION
**Pull request by DaxDupont**
[Changelogs]: 

:cl: Dax Dupont
fix: Supermatter can again blow up again on space tiles.
/:cl:

[why]: Fixes the supermatter not going kaboom on space tiles.
I just moved things around mostly, the only code directly touched is the damage taking for space less since space tile was near instant kaboom.

I also removed excess whitespace.

### CODE REVIEW

File: code/modules/power/supermatter/supermatter.dm [hunk 1]
```diff
@@ -438,10 +435,7 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_shard)
...
 
 			explode()
-
-
-	power -= ((power/500)**3) * powerloss_inhibitor
-
+			
 	return 1
 
 /obj/machinery/power/supermatter_shard/bullet_act(obj/item/projectile/Proj)
```
Comment (DaedalusGame) [inline code/modules/power/supermatter/supermatter.dm]: there's another spaceturf check in bullet_act. Is it worth having a look at that?

File: code/modules/power/supermatter/supermatter.dm [hunk 2]
```diff
@@ -272,107 +272,106 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_shard)
...
+		removed.gases[/datum/gas/oxygen][MOLES] += max(((device_energy + removed.temperature * dynamic_heat_modifier) - T0C) / OXYGEN_RELEASE_MODIFIER, 0)
 
-	removed.temperature = max(0, min(removed.temperature, 2500 * dynamic_heat_modifier))
+		if(produces_gas)
+			env.merge(removed)
+			air_update_turf()
 
-	//Calculate how much gas to release
-	removed.gases[/datum/gas/plasma][MOLES] += max((device_energy * dynamic_heat_modifier) / PLASMA_RELEASE_MODIFIER, 0)
+		for(var/mob/living/carbon/human/l in view(src, HALLUCINATION_RANGE(power))) // If they can see it without mesons on.  Bad on them.
```
Comment (DaedalusGame) [inline code/modules/power/supermatter/supermatter.dm]: Hallucination can probably happen regardless of if we have gases or not?

File: code/modules/power/supermatter/supermatter.dm [hunk 3]
```diff
@@ -272,107 +272,106 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_shard)
...
+			if(!istype(l.glasses, /obj/item/clothing/glasses/meson))
+				var/D = sqrt(1 / max(1, get_dist(l, src)))
+				l.hallucination += power * config_hallucination_power * D
+				l.hallucination = CLAMP(0, 200, l.hallucination)
 
-	removed.gases[/datum/gas/oxygen][MOLES] += max(((device_energy + removed.temperature * dynamic_heat_modifier) - T0C) / OXYGEN_RELEASE_MODIFIER, 0)
+		for(var/mob/living/l in range(src, round((power / 100) ** 0.25)))
```
Comment (DaedalusGame) [inline code/modules/power/supermatter/supermatter.dm]: ditto

File: code/modules/power/supermatter/supermatter.dm [hunk 4]
```diff
@@ -272,107 +272,106 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_shard)
...
-	for(var/mob/living/carbon/human/l in view(src, HALLUCINATION_RANGE(power))) // If they can see it without mesons on.  Bad on them.
-		if(!istype(l.glasses, /obj/item/clothing/glasses/meson))
-			var/D = sqrt(1 / max(1, get_dist(l, src)))
-			l.hallucination += power * config_hallucination_power * D
-			l.hallucination = CLAMP(0, 200, l.hallucination)
-
-	for(var/mob/living/l in range(src, round((power / 100) ** 0.25)))
-		var/rads = (power / 10) * sqrt( 1 / max(get_dist(l, src),1) )
-		l.rad_act(rads)
+		power -= ((power/500)**3) * powerloss_inhibitor
```
Comment (DaedalusGame) [inline code/modules/power/supermatter/supermatter.dm]: Assuming you make it still radiate and hallucinate on space, probably best to also deduct power on space

**Comment by DaedalusGame** [comment]
Looks good now

**Comment by vuonojenmustaturska** [comment]
This probably needs a testmerge, I don't trust this big of a math-heavy diff on the SM.

**Comment by DaxDupont** [comment]
Not math heavy, literally just reordering

**Comment by DaedalusGame** [comment]
This is true, there's no math changes aside from the damage formula when on space.

**Comment by DaxDupont** [comment]
Not only that, the old calculations caused it to be damaged to explosion damage in 20-30 seconds by removing the tile and if you put that tile back it exploded.

**Pull request merged**